stat/spatial/spatial.go

```diff
@@ -43,11 +42,21 @@ func GetisOrdGStar(i int, data, weights []float64, locality mat.Matrix) float64
 	n := float64(len(data))
 	mean, std := stat.MeanStdDev(data, weights)
 	var dwd, dww, sw float64
-	for j, v := range data {
-		w := locality.At(i, j)
-		sw += w
-		dwd += w * v
-		dww += w * w
+	if band, ok := locality.(mat.Banded); ok {
```

Comment:
(actually, interestingly, that taurus structure could be done with the blas.Banded layout except actually using the `*` locations)